lib/p2p/Pool.ts

```diff
@@ -245,11 +245,11 @@ class Pool extends EventEmitter {
    * If the node is banned, already connected, or has no listening addresses, then do nothing.
    * Additionally, if we're already trying to connect to a given node also do nothing.
    * @param nodes a collection of nodes with a `forEach` iterator to attempt to connect to
-   * @param ignoreKnown whether to ignore nodes we are already aware of, defaults to false
+   * @param allowKnown whether to allow nodes we are already aware of, defaults to true
    * @param retryConnecting whether to attempt retry connecting, defaults to false
    * @returns a promise that will resolve when all outbound connections resolve
    */
-  private connectNodes = (nodes: NodeConnectionIterator, ignoreKnown = false, retryConnecting = false) => {
+  private connectNodes = (nodes: NodeConnectionIterator, allowKnown = true, retryConnecting = false) => {
```

Comment:
I'm curious where the bug fix pertaining to discovery actually is. The code here makes sense to me (and is easier to understand than before), but it seems to function the same as before. Could you explain a bit more how it is fixed?

Reply:
The previous condition was:
````
const isNotIgnored = this.nodes.has(node.nodePubKey) && !ignoreKnown;
````

It’s checking whether we are not ignoring this node, which logically applies only to nodes we already know (the first condition). But if we simply don't know this node, it will return false, and we won't attempt to connect.

Reply:
Ahhhh ok, yes that makes sense thank you. It's clearer now (and working as intended).